src/NUnitFramework/framework/Constraints/EqualNumericWithoutUsingConstraint.cs

```diff
@@ -141,12 +141,17 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
             {
                 hasSucceeded = false;
             }
-            else if (actual is T t)
+            else if (actual is T t && (typeof(T).IsPrimitive || typeof(T) == typeof(decimal)))
```

Comment:
Thanks for catching the `decimal` type here.

Reply:
I didn't, but our unit tests did!